### PR TITLE
US020: Reservation Action Added

### PR DESCRIPTION
**Pull request by SamDotNetDev**

### CODE REVIEW

File: ParkingZoneApp/Models/Reservation.cs [hunk 1]
```diff
@@ -22,7 +22,10 @@ public class Reservation
         public virtual ParkingSlot ParkingSlot { get; set; }
 
         [Required]
-        public int UserId { get; set; }
+        public string VehicleNumber { get; set; }
+
+        [Required]
+        public string UserId { get; set; }
```
Comment (dilmurod-rozikov) [inline ParkingZoneApp/Models/Reservation.cs:28]: public string UserId { get; set; }
does this userId belong to ApplicationUser if yes I guess it should be type of Guid not string

File: ParkingZoneApp/Views/Reservation/Reserve.cshtml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    <div class="row">
+        <h2>@Model.ParkingZoneName , @Model.ParkingZoneAddress</h2>
+        <h6>Slot Number: @Model.ParkingSlotNumber</h6>
+        <hr />
+        <div class="col-md-4">
+            <form asp-action="Reserve">
+                <div asp-validation-summary="ModelOnly" class="text-danger"></div>
+
+                <input type="hidden" asp-for="ParkingSlotId" />
+
```
Comment (dilmurod-rozikov) [inline ParkingZoneApp/Views/Reservation/Reserve.cshtml]: space

File: ParkingZoneApp/Views/Reservation/Reserve.cshtml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+@if (string.IsNullOrEmpty(ViewBag.SuccessMessage))
+{
+    <div class="row">
+        <h2>@Model.ParkingZoneName , @Model.ParkingZoneAddress</h2>
+        <h6>Slot Number: @Model.ParkingSlotNumber</h6>
+        <hr />
+        <div class="col-md-4">
+            <form asp-action="Reserve">
+                <div asp-validation-summary="ModelOnly" class="text-danger"></div>
+
```
Comment (dilmurod-rozikov) [inline ParkingZoneApp/Views/Reservation/Reserve.cshtml]: scpace

File: Tests/ReservationTests/Controller/ReservationControllerTests.cs [hunk 4]
```diff
@@ -72,5 +77,151 @@ public void GivenFreeSlotsVM_WhenFreeSlotsIsCalled_ThenReturnsViewResult()
...
+
+            //Assert
+            Assert.NotNull(result);
+            var model = Assert.IsType<ViewResult>(result).Model;
+            Assert.IsType<ReserveVM>(model);
+            _slotService.Verify(x => x.GetById(Id), Times.Once);
+        }
+
+        [Fact]
+        public void GivenReserveViewModel_WhenReserveIsCalledToPost_ThenReturnsNotFound()
```
Comment (dilmurod-rozikov) [inline Tests/ReservationTests/Controller/ReservationControllerTests.cs]: `public void `GivenReserveViewModel_WhenReserveIsCalledToPost_ThenReturnsNotFound()`

better version would be: 
        `public` void `GivenReserveViewModel_WhenPostReserveCalled_ThenReturnsNotFound()`

File: Tests/ReservationTests/Controller/ReservationControllerTests.cs [hunk 5]
```diff
@@ -72,5 +77,151 @@ public void GivenFreeSlotsVM_WhenFreeSlotsIsCalled_ThenReturnsViewResult()
...
+            var result = _controller.Reserve(vm);
+
+            //Assert
+            Assert.NotNull(result);
+            Assert.IsType<NotFoundResult>(result);
+            _slotService.Verify(x => x.GetById(vm.ParkingSlotId), Times.Once);
+        }
+
+        [Fact]
+        public void GivenReserveViewModel_WhenReserveIsCalledToPost_ThenModelStateIsFalseAndReturnsViewResult()
```
Comment (dilmurod-rozikov) [inline Tests/ReservationTests/Controller/ReservationControllerTests.cs]: same as above

**Review by dilmurod-rozikov — CHANGES_REQUESTED** [review]

File: ParkingZoneApp/Models/Reservation.cs (hunk 1 above)
Comment (SamDotNetDev) [reply ParkingZoneApp/Models/Reservation.cs:28]: image
in Controller here to get userId Im using this method and it returns string if i do UserId property Guid the function is giving exception.

**Review by dilmurod-rozikov — APPROVED** [review]

**Review by abdullokhtolibjonov — APPROVED** [review]

**Pull request merged**